src/app/d2-loadout-builder/generated-sets/GeneratedSetItem.tsx

```diff
@@ -1,17 +1,31 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import { DimPlug, DimItem } from '../../inventory/item-types';
 import LoadoutBuilderItem from '../LoadoutBuilderItem';
 import { LockedItemType } from '../types';
 import ItemSockets from '../../item-popup/ItemSockets';
-import { statHashes } from '../process';
 import _ from 'lodash';
 import styles from './GeneratedSetItem.m.scss';
 import { AppIcon } from 'app/shell/icons';
-import { faRandom, faUnlock } from '@fortawesome/free-solid-svg-icons';
+import { faRandom, faLock } from '@fortawesome/free-solid-svg-icons';
 import { showItemPicker } from 'app/item-picker/item-picker';
 import { t } from 'app/i18next-t';
 import { lockedItemsEqual } from './utils';
+import { generateMixesFromPerks } from '../process';
 
+/**
+ * Figure out which (if any) non-selected perks should be selected to get the chosen stat mix.
+ */
+function identifyAltPerkChoicesForChosenStats(item: DimItem, chosenValues: number[]) {
```

Comment:
dang, this is next level haha. nice to keep it wrapped in here though versus handling the 'base'/'bonus'/'value' stats.